File: 6 kyu/Your order, please/solution.jl
function order(words)
  findnum(w) = match(r"\d", w).match
  join(sort(split(words, " "), lt = (a, b) -> findnum(a) < findnum(b)), " ")
end

_____________________________________________
function order(words)
  join(sort(split(words), by = x-> filter(isdigit, x)), " ")
end

_____________________________________________
function order(words)
    join(sort!(split(words), by=numberWord), " ")
end

function numberWord(word)
    for ch in word
        if isdigit(ch)
            return ch - '0'
        end
    end
    0
end

_____________________________________________
function order(words)
  join(map(((i, word),) -> word, sort(map(word -> (match(r"\d", word).match, word), split(words)))), " ")
end

_____________________________________________
function order(words)
  sorted = fill("", length(split(words)))
  for word in split(words)
    for i in word
      try 
        n = parse(Int, i)
        println(typeof(n))
        sorted[n] = word
      catch
        continue
      end
    end
  end
  return join(sorted, " ")
end

File: 6 kyu/Your order, please/solution.cs
using System;
using System.Linq;

public static class Kata
{
  public static string Order(string words)
        {
            if (string.IsNullOrEmpty(words)) return words;
            return string.Join(" ", words.Split(' ').OrderBy(s => s.ToList().Find(c => char.IsDigit(c))));
        }
}

_____________________________________________
using System;
using System.Linq;

public static class Kata
{
  public static string Order(string words)
  {
    return string.Join(" ", words.Split().OrderBy(w => w.SingleOrDefault(char.IsDigit)));
  }
}

_____________________________________________
using System;
using System.Text;
using System.Linq;
using System.Text.RegularExpressions;
public static class Kata
{
  public static string Order(string words)
  {
    if (words == null) return words;
    var orderedWords = words.Split(" ")
                            .OrderBy(x => Regex.Match(x, @"\d").Value);
                      
    return string.Join(" ", orderedWords);
  }
}

_____________________________________________
using System;

public static class Kata
{
  public static string Order(string words)
  {    
    if (words == "")
      return words;
    
    string[] splited = words.Split(' ');  
    string result = "";
    for (int i = 1; i < 10; i++)
      {
        foreach (string word in splited)
          {
            if (word.Contains(Convert.ToString(i)))
            {   
              if (result == "")
                result += word;
              else  
                result += " " + word;
            }
          }
      }
    return result;
  }
}

_____________________________________________
using System;
using System.Linq;

public static class Kata
{
  public static string Order(string words)
  {
    if (words == "")
            return words;
        var stringArray = words.Split(" ");
        var returnString = new string[stringArray.Length];
        foreach (var i in stringArray)
            returnString[i.Where(n => Char.IsDigit(n)).First() - '1'] = i;
        return String.Join(" ", returnString);
  }
}
